src/clang.rs

```diff
@@ -809,27 +802,65 @@ impl Comment {
         }
     }
 
-    /// Given that this comment is an HTML start tag, get the number of HTML
-    /// attributes it has.
-    pub fn get_num_tag_attrs(&self) -> c_uint {
-        unsafe {
-            clang_HTMLStartTag_getNumAttrs(self.x)
+    /// Given that this comment is an HTML start tag, get its attributes.
+    pub fn get_tag_attrs(&self) -> CommentAttributesIntoIterator {
+        CommentAttributesIntoIterator {
+            x: self.x,
+            length: unsafe { clang_HTMLStartTag_getNumAttrs(self.x) },
+            index: 0
         }
     }
+}
 
-    /// Given that this comment is an HTML start tag, get the `idx`th
-    /// attribute's name.
-    pub fn get_tag_attr_name(&self, idx: c_uint) -> String {
-        unsafe {
-            String_ { x: clang_HTMLStartTag_getAttrName(self.x, idx) }.to_string()
+/// An iterator for a comment's children
+pub struct CommentChildrenIntoIterator {
+    parent: CXComment,
+    length: c_uint,
+    index: c_uint
+}
+
+impl Iterator for CommentChildrenIntoIterator {
+    type Item = Comment;
+    fn next(&mut self) -> Option<Comment> {
+        if self.index < self.length {
+            let idx = self.index;
+            self.index += 1;
+            Some( Comment { x: unsafe { clang_Comment_getChild(self.parent, idx) } } )
+        }
+        else {
+            None
         }
     }
+}
 
-    /// Given that this comment is an HTML start tag, get the `idx`th
-    /// attribute's value.
-    pub fn get_tag_attr_value(&self, idx: c_uint) -> String {
-        unsafe {
-            String_ { x: clang_HTMLStartTag_getAttrValue(self.x, idx) }.to_string()
+/// An HTML start tag comment attribute
+pub struct CommentAttribute {
+    /// HTML start tag attribute name
+    pub name: String,
+    /// HTML start tag attribute value
+    pub value: String
+}
+
+/// An iterator for a comment's attributes
+pub struct CommentAttributesIntoIterator {
```

Comment:
Same, `CommentAttributesIterator` seems like a better name for this, since it doesn't consume the comment.